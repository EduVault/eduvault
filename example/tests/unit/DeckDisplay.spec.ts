import { mount, createLocalVue } from '@vue/test-utils';
import DeckDisplay from '@/components/DeckDisplay.vue';
import { library } from '@fortawesome/fontawesome-svg-core';
import {
  faTrashAlt,
  faPlusSquare,
  faCheck,
  faEdit,
  faTimes,
} from '@fortawesome/free-solid-svg-icons';
import {
  FontAwesomeIcon,
  //  FontAwesomeLayers
} from '@fortawesome/vue-fontawesome';
const localVue = createLocalVue();
library.add(faTrashAlt, faPlusSquare, faCheck, faEdit, faTimes);
localVue.component('fontAwesomeIcon', FontAwesomeIcon);

describe('DeckDisplay.vue', () => {
  const wrapper = mount(DeckDisplay, {
    localVue,
    propsData: {
      deck: {
        _id: '12f14f23',
        cards: [
          {
            _id: '1',
            frontText: 'FrontText',
            backText: 'BackText',
          },
          {
            _id: '2',
            frontText: 'FrontText',
            backText: 'BackText',
          },
        ],
        title: 'Title',
      },
    },
  }) as any;

  it('renders', () => {
    expect(wrapper.exists()).toBe(true);
  });
  it('matches snapshot', () => {
    expect(wrapper.element).toMatchSnapshot();
  });
  it('has font awesome icon buttons', () => {
    const buttons = wrapper.findAllComponents(FontAwesomeIcon);
    // this works in mount but not shallowMount
    expect(buttons).toHaveLength(4);
  });
  it('displays deck title', () => {
    const title = wrapper.find('.deck-display__title');
    expect(title.text()).toBe('Title');
  });
  it('displays correct number of cards', () => {
    const cards = wrapper.findAll('.deck-display__card-container');
    expect(cards).toHaveLength(2);
  });
  it('emits the correct data when edit card is clicked', async () => {
    const editButton = wrapper.findAll('.buttons-col__button--edit').at(0);
    await editButton.trigger('click');
    const expectedOutput = {
      frontText: 'FrontText',
      backText: 'BackText',
      _id: '1',
      deckId: 'Deck Id',
    };
    expect(JSON.stringify(wrapper.emitted().openEditor[0][0])).toBe(JSON.stringify(expectedOutput));
  });
  it('emits the correct data when delete card is clicked', async () => {
    const deleteButton = wrapper.findAll('.buttons-col__button--delete').at(0);
    await deleteButton.trigger('click');
    const expectedOutput = {
      _id: '1',
      deckId: 'deckId',
    };
    expect(JSON.stringify(wrapper.emitted().deleteCard[0][0])).toBe(JSON.stringify(expectedOutput));
  });
});
