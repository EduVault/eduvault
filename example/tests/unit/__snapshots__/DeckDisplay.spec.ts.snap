// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeckDisplay.vue matches snapshot 1`] = `
<div
  class="deck-display"
>
  <h2
    class="deck-display__title title"
  >
    Title
  </h2>
   
  <div
    class="deck-display__card-container card-container"
  >
    <div
      class="flashcard deck-display__flashcard"
    >
      <div
        class="flashcard__wrapper a-flipInX flashcard__front"
        style="background-color: white; color: black;"
      >
        <div
          class="flashcard__header"
          style="padding-bottom: 15px;"
        >
          Question
        </div>
         
        <div
          class="flashcard__content"
        >
          <p
            class="flashcard__text"
            style="font-size: 2em; font-weight: bold;"
          >
            
        FrontText
      
          </p>
           
          <!---->
        </div>
         
        <div
          class="flashcard__footer"
        >
          Click to flip
        </div>
      </div>
       
      <div
        class=" flashcard__wrapper a-flipInX flashcard__back"
        style="background-color: #42b983; color: white; display: none;"
      >
        <div
          class="flashcard__header"
          style="padding-bottom: 15px;"
        >
          Answer
        </div>
         
        <div
          class="flashcard__content"
        >
          <p
            class="flashcard__text"
            style="font-size: 2em; font-weight: bold;"
          >
            
        BackText
      
          </p>
           
          <!---->
        </div>
         
        <div
          class="flashcard__footer"
        >
          Click to flip
        </div>
      </div>
    </div>
     
    <div
      class="deck-display__buttons-col buttons-col"
    >
      <svg
        aria-hidden="true"
        class="buttons-col__button buttons-col__button--edit button svg-inline--fa fa-edit fa-w-18 fa-2x"
        data-icon="edit"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          fill="currentColor"
        />
      </svg>
       
      <svg
        aria-hidden="true"
        class="buttons-col__button buttons-col__button--delete button svg-inline--fa fa-trash-alt fa-w-14 fa-2x"
        data-icon="trash-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
  <div
    class="deck-display__card-container card-container"
  >
    <div
      class="flashcard deck-display__flashcard"
    >
      <div
        class="flashcard__wrapper a-flipInX flashcard__front"
        style="background-color: white; color: black;"
      >
        <div
          class="flashcard__header"
          style="padding-bottom: 15px;"
        >
          Question
        </div>
         
        <div
          class="flashcard__content"
        >
          <p
            class="flashcard__text"
            style="font-size: 2em; font-weight: bold;"
          >
            
        FrontText
      
          </p>
           
          <!---->
        </div>
         
        <div
          class="flashcard__footer"
        >
          Click to flip
        </div>
      </div>
       
      <div
        class=" flashcard__wrapper a-flipInX flashcard__back"
        style="background-color: #42b983; color: white; display: none;"
      >
        <div
          class="flashcard__header"
          style="padding-bottom: 15px;"
        >
          Answer
        </div>
         
        <div
          class="flashcard__content"
        >
          <p
            class="flashcard__text"
            style="font-size: 2em; font-weight: bold;"
          >
            
        BackText
      
          </p>
           
          <!---->
        </div>
         
        <div
          class="flashcard__footer"
        >
          Click to flip
        </div>
      </div>
    </div>
     
    <div
      class="deck-display__buttons-col buttons-col"
    >
      <svg
        aria-hidden="true"
        class="buttons-col__button buttons-col__button--edit button svg-inline--fa fa-edit fa-w-18 fa-2x"
        data-icon="edit"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 576 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M402.6 83.2l90.2 90.2c3.8 3.8 3.8 10 0 13.8L274.4 405.6l-92.8 10.3c-12.4 1.4-22.9-9.1-21.5-21.5l10.3-92.8L388.8 83.2c3.8-3.8 10-3.8 13.8 0zm162-22.9l-48.8-48.8c-15.2-15.2-39.9-15.2-55.2 0l-35.4 35.4c-3.8 3.8-3.8 10 0 13.8l90.2 90.2c3.8 3.8 10 3.8 13.8 0l35.4-35.4c15.2-15.3 15.2-40 0-55.2zM384 346.2V448H64V128h229.8c3.2 0 6.2-1.3 8.5-3.5l40-40c7.6-7.6 2.2-20.5-8.5-20.5H48C21.5 64 0 85.5 0 112v352c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48V306.2c0-10.7-12.9-16-20.5-8.5l-40 40c-2.2 2.3-3.5 5.3-3.5 8.5z"
          fill="currentColor"
        />
      </svg>
       
      <svg
        aria-hidden="true"
        class="buttons-col__button buttons-col__button--delete button svg-inline--fa fa-trash-alt fa-w-14 fa-2x"
        data-icon="trash-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class=""
          d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;
